sci-libs/dsdp/dsdp-5.8-r3.ebuild

```diff
@@ -0,0 +1,79 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit multilib toolchain-funcs versionator
+
+MYP=DSDP${PV}
+
+DESCRIPTION="Software for interior-point for semidefinite programming"
+HOMEPAGE="http://www.mcs.anl.gov/hs/software/DSDP/"
+SRC_URI="http://www.mcs.anl.gov/hs/software/DSDP//${MYP}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos"
+IUSE="doc examples"
+
+RDEPEND="virtual/lapack"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+S="${WORKDIR}/${MYP}"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-readsdpa.patch
+	"${FILESDIR}"/${P}-malloc.patch
+	"${FILESDIR}"/${P}-gold.patch
+)
+
+make_shared_lib() {
+	local soname=$(basename "${1%.a}")$(get_libname $(get_major_version))
+	einfo "Making ${soname}"
+	${2:-$(tc-getCC)} ${LDFLAGS}  \
+		-shared -Wl,-soname="${soname}" \
+		$([[ ${CHOST} == *-darwin* ]] && echo "-Wl,-install_name -Wl,${EPREFIX}/usr/$(get_libdir)/${soname}") \
+		-Wl,--whole-archive "${1}" -Wl,--no-whole-archive \
+		-o $(dirname "${1}")/"${soname}" \
+		-lm $($(tc-getPKG_CONFIG) --libs blas lapack) || return 1
+}
+
+src_prepare() {
+	default
+	# to do proper parallel compilation
+	find . -name Makefile -exec \
+		 sed -i -e 's:make :$(MAKE) :g' '{}' \;
```

Comment:
Works. After src_repare, old version and new versions of ${TMPDIR} agree (except environment)